src/components/Tag/Tabs/ImageSelector.jsx

```diff
@@ -0,0 +1,395 @@
+import { useSearchParams } from 'react-router-dom';
+import { debounce, isEmpty } from 'lodash';
+import { useCombobox } from 'downshift';
+import { HEADER_SEARCH_PAGE_SIZE } from 'utilities/paginationConstants';
+
+const useStyles = makeStyles(() => ({
+  searchContainer: {
+    display: 'inline-block',
+    backgroundColor: '#FFFFFF',
+    boxShadow: '0 0.313rem 0.625rem rgba(131, 131, 131, 0.08)',
+    borderRadius: '0.625rem',
+    position: 'relative',
+    zIndex: 1150
+  },
+  searchContainerFocused: {
+    backgroundColor: '#FFFFFF'
+  },
+  search: {
+    position: 'relative',
+    flexDirection: 'row',
+    boxShadow: '0rem 0.3125rem 0.625rem rgba(131, 131, 131, 0.08)',
+    border: '0.063rem solid #8A96A8',
+    borderRadius: '0.625rem',
+    zIndex: 1155
+  },
+  searchFocused: {
+    border: '0.125rem solid #E0E5EB',
+    backgroundColor: '#FFFFF'
+  },
+  searchFailed: {
+    border: '0.125rem solid #ff0303'
+  },
+  resultsWrapper: {
+    margin: '0',
+    marginTop: '-5%',
+    paddingTop: '5%',
+    position: 'absolute',
+    alignItems: 'center',
+    justifyContent: 'center',
+    backgroundColor: '#FFFFFF',
+    boxShadow: '0rem 0.3125rem 0.625rem rgba(131, 131, 131, 0.08)',
+    borderBottomLeftRadius: '0.625rem',
+    borderBottomRightRadius: '0.625rem',
+    // border: '0.125rem solid #E7E7E7',
+    borderTop: 0,
+    width: '100%',
+    overflowY: 'auto',
+    zIndex: 1
+  },
+  resultsWrapperFocused: {
+    backgroundColor: '#FFFFFF'
+  },
+  resultsWrapperHidden: {
+    display: 'none'
+  },
+  input: {
+    marginLeft: 1,
+    width: '90%',
+    paddingLeft: 10,
+    height: '40px',
+    fontSize: '1rem',
+    backgroundColor: '#FFFFFF',
+    borderRadius: '0.625rem',
+    color: '#8A96A8'
+  },
+  inputFocused: {
+    backgroundColor: '#FFFFFF',
+    borderRadius: '0.625rem',
+    color: 'rgba(0, 0, 0, 0.6);'
+  },
+  searchItem: {
+    alignItems: 'center',
+    justifyContent: 'flex-start',
+    color: '#000000',
+    height: '2.75rem',
+    padding: '0 5%',
+    cursor: 'pointer'
+  },
+  searchItemIconBg: {
+    backgroundColor: '#FFFFFF',
+    height: '1.5rem',
+    width: '1.5rem',
+    display: 'flex',
+    justifyContent: 'center',
+    alignItems: 'center',
+    overflow: 'hidden'
+  },
+  searchItemIcon: {
+    color: '#0000008A',
+    minHeight: '100%',
+    minWidth: '100%',
+    objectFit: 'fill'
+  }
+}));
+
+function ImageSelector({ setSearchCurrentValue = () => {}, name, tag }) {
+  // digest, selectedPlatform
+  const [inputHelpText, setInputHelpText] = useState('Specify a repo:tag');
+  const [activePlatformSelection, setActivePlatformSelection] = useState(false);
+  const [platformOptions, setPlatformOptions] = useState([]);
+  const [searchQuery, setSearchQuery] = useState('');
+  const [platform, setPlatform] = useState('');
+  const [suggestionData, setSuggestionData] = useState([]);
+  const [queryParams] = useSearchParams();
+  const search = queryParams.get('search') || '';
```

Comment:
similar story, do we want the query string to control the value of this search component?